### ISSUE TITLE
[no-irregular-whitespace] add option for allow ignore check on some char

### ISSUE THREAD

**Issue by bluelovers** [eslint/eslint#11781]
**What rule do you want to change?**

no-irregular-whitespace
https://cn.eslint.org/docs/rules/no-irregular-whitespace

add option for allow ignore check on some char

this make we can enable this rule, but no need use too much on some char we use offen on current code

`// eslint-disable-next-line no-irregular-whitespace`
or 
`　` => `\u3000`

**Does this change cause the rule to produce more or fewer warnings?**

fewer warnings

**How will the change be implemented? (New option, new default behavior, etc.)?**

new options
```
{
  ignore: '\u3000'
}
```

**Please provide some example code that this change will affect:**

```js
/[ \t\uFEFF\xA0　]+$/
```

**What does the rule currently do for this code?**

can't ignore some char

**What will the rule do after it's changed?**

allow ignore some char

**Are you willing to submit a pull request to implement this change?**

not sure, because im 0% about how to read source code about eslint

**Comment by platinumazure** [comment]
Hi @bluelovers, thanks for the issue.

I'm not sure I understand the use case. Why do you need to ignore some characters but not others? Are there no other ways to represent the characters?

**Comment by bluelovers** [comment]
because human type full width space  better than `\u3000`

**Comment by bluelovers** [comment]
if can, hope someone can tell me, why it fail lol when i try create a plugin for this
https://stackoverflow.com/questions/56404119/eslint-why-my-plugin-show-definition-for-rule-was-not-found

**Comment by mysticatea** [comment]
Unicode has a lot of whitespace characters. We are hard to distinguish those whitespace characters because those are invisible. On the other hand, `"\u3000"` is clear that it's Ideographic Space because it's visible.

I know that we can input Ideographic Space character very easy (e.g., <kbd>Shift</kbd>+<kbd>Space</kbd> in Japan), But, in general, we should consider that reading is important rather than writing in order to maintain the software.

Therefore, I think the `ignores` option is opposing the purpose of the rule.

**Comment by platinumazure** [comment]
Likewise, I am not sure I understand the purpose of this proposal either (in that I don't understand the benefit, as compared to the potential cost).

**Comment by bluelovers** [comment]
nvm, can anyone tell me why my plugin is show error message?

**Comment by kaicataldo** [comment]
@bluelovers Can you create a new repository that reproduces the error you’re seeing so that we can try to help you resolve this?

**Comment by bluelovers** [comment]
[err-test.zip](https://github.com/eslint/eslint/files/3269079/err-test.zip)

```
G:\Users\The Project\temp\err-test\index.ts
  1:1   error    Definition for rule '@bluelovers/no-irregular-whitespace' was not found  @bluelovers/no-irregular-whitespace
  5:19  warning  Missing semicolon                                                        @typescript-eslint/semi

✖ 2 problems (1 error, 1 warning)
  0 errors and 1 warning potentially fixable with the `--fix` option.

```
